src/main/java/com/amazon/opendistroforelasticsearch/security/securityconf/DynamicConfigFactory.java

```diff
@@ -230,7 +238,7 @@ public void registerDCFListener(DCFListener listener) {
     }
     
     public static interface DCFListener {
-        void onChanged(ConfigModel cm, DynamicConfigModel dcm, InternalUsersModel ium);
+        void onChanged(ConfigModel cm, DynamicConfigModel dcm, InternalUsersModel ium, NodesDnModel nm);
```

Comment:
I don't think that changing the interface is well justified. Only `DefaultInterClusterRequestEvaluator` is interested in `NodesDnModel`, all other classes that implement `DCFListener` actually do not use `NodesDnModel` on the callback event.

Reply:
The behavior in opendistroforelasticsearch/security for older versions (0.9 and 0.10) had support for subscribing and listening to changes for specific type.

However that was removed in later versions where irrespective of type that changed, listener receives everything - https://github.com/opendistro-for-elasticsearch/security/blob/master/src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/ConfigurationRepository.java#L294-L298

I'll give this another thought, but do you have any suggestion?

Reply:
Please see #364

Reply:
> Please see #364

I left a comment on yours. May be Hardik has some context on the behavior change. But I an adapt to your change quickly once it is pushed

Reply:
Adapted to pub-sub changes in master.